pmlc/conversion/tile_to_pxa/tests/eltwise_add.mlir

```diff
@@ -17,3 +17,56 @@ func @eltwise_add(
 // CHECK: affine.load
 // CHECK: addf
 // CHECK: affine.store
+
+func @eltwise_add_f32_index(%arg0: tensor<4x1xf32>) -> (tensor<4x1xf32>) {
+  %c7 = tile.constant 7
+  %1 = "eltwise.add"(%arg0, %c7) : (tensor<4x1xf32>, index) -> tensor<4x1xf32>
+  return %1 : tensor<4x1xf32>
+}
+
+// CHECK-LABEL: func @eltwise_add_f32_index
+// CHECK: affine.load
+// CHECK-NEXT: sitofp {{.*}} i64 to f32
```

Comment:
I think we should also check for the index cast

Reply:
Canonicalizer is rezer is removing the index_cast. It makes sense since the value is constant - 7 - and the i64 type is happily fitting this constant with no propblem.
Code before canonicalizer...
func @eltwise_add_f32_index(%arg0: memref<4x1xf32>, %arg1: memref<4x1xf32>) {
    %c7 = constant 7 : index
    %0 = alloc() : memref<4x1xf32>
    affine.parallel (%arg2, %arg3) = (0, 0) to (4, 1) {
      %c0 = constant 0 : index
      %1 = affine.load %arg0[%arg2, %c0] : memref<4x1xf32>
      %2 = index_cast %c7 : index to i64
      %3 = sitofp %2 : i64 to f32
      %4 = addf %1, %3 : f32
      affine.store %4, %arg1[%arg2, %arg3] : memref<4x1xf32>
Code after canonicalizer...
func @eltwise_add_f32_index(%arg0: memref<4x1xf32>, %arg1: memref<4x1xf32>) {
    %c7_i64 = constant 7 : i64
    affine.parallel (%arg2, %arg3) = (0, 0) to (4, 1) {
      %0 = affine.load %arg0[%arg2, 0] : memref<4x1xf32>
      %1 = sitofp %c7_i64 : i64 to f32
      %2 = addf %0, %1 : f32
      affine.store %2, %arg1[%arg2, %arg3] : memref<4x1xf32>
    }
    return
  }

Reply:
OK, looks good then.